File: frontend/src/types/activity-type.ts
export interface ActivityType {
  id: string;
  name: string;
  is_active: boolean;
  unit_type: string;
  default_rate?: number;
}

export interface CreateActivityType {
  name: string;
  unit_type?: string;
  default_rate?: number;
}

export interface UpdateActivityType {
  name: string;
  is_active: boolean;
  unit_type?: string;
  default_rate?: number | null;
}

File: backend/crates/konto-db/src/entities/activity_type.rs
use sea_orm::entity::prelude::*;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Serialize, Deserialize)]
#[sea_orm(table_name = "activity_types")]
pub struct Model {
    #[sea_orm(primary_key, auto_increment = false)]
    pub id: String,
    #[sea_orm(unique)]
    pub name: String,
    pub is_active: bool,
    pub unit_type: String,
    #[sea_orm(column_type = "Decimal(Some((10, 2)))", nullable)]
    pub default_rate: Option<rust_decimal::Decimal>,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {}

impl ActiveModelBehavior for ActiveModel {}
